habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/util/ContainerizeDepsDockerfileHelper.java

```
@@ -37,158 +35,215 @@ public ContainerizeDepsDockerfileHelper(ContainerizeDepsMojo containerizeDepsMoj
+            String dockerfileContent = Files.readString(Paths.get(containerizeDepsMojo.getDockerfile().toString()));
+            if (!builderContentPresent(dockerfileContent)) {
+                if (!addBuilderContentCommentPresent(dockerfileContent)) {
+                    return getWrappedBuilderStageContent(true) + "\n" + dockerfileContent;
+                } else {
+                    return dockerfileContent.replace(HABUSHU_BUILDER_STAGE, getWrappedBuilderStageContent(true));
                 }
-                line = buffer.readLine();
-            }
-            if (!builderStageContentIncluded) {
-                content.insert(Integer.max(firstFromLine, 0), wrapWithHabushuComment(builderStageContent, HABUSHU_BUILDER_STAGE) + "\n\n");
-            }
-            if (!finalStageContentIncluded) {
-                content.append("\n");
-                content.append(wrapWithHabushuComment(finalStageContent, HABUSHU_FINAL_STAGE)).append("\n");
             }
+            return dockerfileContent;
         } catch (IOException e) {
-            throw new HabushuException("Could not update Dockerfile with container stage logic.", e);
+            throw new HabushuException("Could not update Dockerfile with builder stage logic.", e);
         }
-        return content.toString();
     }
 
-    private static String wrapWithHabushuComment(String content, String stage) {
-        StringBuilder contentBuilder = new StringBuilder();
-        contentBuilder.append(stage).append(HABUSHU_COMMENT_START).append("\n");
-        contentBuilder.append(content).append("\n");
-        contentBuilder.append(stage).append(HABUSHU_COMMENT_END);
-        return contentBuilder.toString();
+    private String updateDockerfileWithFinalLogic(String dockerfileContent) {
+        if (!finalContentPresent(dockerfileContent)) {
+            if (!addFinalContentCommentPresent(dockerfileContent)) {
+                return dockerfileContent + "\n\n" + getWrappedFinalStageContent();
```

Q: Should we raise an error here instead since we don't know what the dockerfileContent looks like?